### ISSUE TITLE
[Ballista] Custom object stores only work for CSV files

### ISSUE THREAD

**Issue by andygrove** [apache/arrow-datafusion#2546]
**Describe the bug**
The implementation of  `read_csv` creates `ListingTableConfig` using the URI rather than the path of the file (so `test://employee.csv` rather than `employee.csv`) and this is required to make `roundtrip_logical_plan_custom_ctx` pass.

There are no tests for custom contexts for other file types and those paths use path instead of URI, so don't seem to support custom object stores.

There is some code duplicated between all of these read methods that could probably be refactored into common helper functions.

**To Reproduce**
Try implementing a version of `roundtrip_logical_plan_custom_ctx` for each file type.

**Expected behavior**
Functionality should be consistent between file types.

**Additional context**
None

**Comment by andygrove** [comment]
@thinkharderdev @mingmwang I could use some help with this if you have time to look.

**Comment by thinkharderdev** [comment]
Yeah, the issue here is that the serde logic uses the file scheme to resolve the `ObjectStore`. So we were relying on the `test` scheme to resolve the `TestObjectStore`. Why would we not serialize the entire file URI?
